Geometry/GEMGeometryBuilder/plugins/BuildFile.xml

```diff
@@ -1,18 +1,25 @@
-<use name="Geometry/MuonNumbering"/>
-<use name="Geometry/Records"/>
-<use name="Geometry/GEMGeometry"/>
-<use name="Geometry/GEMGeometryBuilder"/>
-<use name="FWCore/Framework"/>
-<use name="FWCore/ParameterSet"/>
+<use name="CondFormats/AlignmentRecord"/>
 <use name="CondFormats/Alignment"/>
 <use name="CondFormats/DataRecord"/>
 <use name="CondFormats/GeometryObjects"/>
+<use name="FWCore/Framework"/>
+<use name="FWCore/ParameterSet"/>
+<use name="Geometry/MuonNumbering"/>
+<use name="Geometry/Records"/>
+
+<library name="GEMGeometryPlugins" file="GEMGeometryESModule.cc">
 <use name="DetectorDescription/Core"/>
-<library file="GEMGeometryESModule.cc"> 
-  <flags EDM_PLUGIN="1"/>
-</library>
-<library file="ME0GeometryESModule.cc"> 
```

Comment:
@slomeo - the ME0 geometry builder is dropped... Please, restore.

Reply:
@ianna    There is a problem if I restore the ME0 Geometry Builder in  Geometry/GEMGeometryBuilder/plugins/BuildFile.xml even if I made a "scram b clean" and then a "scram b -j 10". 
The messages I see is:
....
>> Building shared library tmp/slc7_amd64_gcc820/src/DetectorDescription/DDCMS/plugins/DetectorDescriptionDD4HepPlugins/libDetectorDescriptionDD4HepPlugins.so
/cvmfs/cms-ib.cern.ch/nweek-02613/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: tmp/slc7_amd64_gcc820/src/Geometry/GEMGeometryBuilder/plugins/ME0GeometryESModule/ME0GeometryESModule.cc.o: in function `ME0GeometryESModule::produce(MuonGeometryRecord const&)':
ME0GeometryESModule.cc:(.text+0x3f7): undefined reference to `ME0GeometryBuilderFromDDD::ME0GeometryBuilderFromDDD()'
/cvmfs/cms-ib.cern.ch/nweek-02613/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: ME0GeometryESModule.cc:(.text+0x413): undefined reference to `ME0GeometryBuilderFromDDD::build(DDCompactView const*, MuonDDDConstants const&)'
/cvmfs/cms-ib.cern.ch/nweek-02613/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: ME0GeometryESModule.cc:(.text+0x41e): undefined reference to `ME0GeometryBuilderFromDDD::~ME0GeometryBuilderFromDDD()'
/cvmfs/cms-ib.cern.ch/nweek-02613/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: ME0GeometryESModule.cc:(.text+0x5fa): undefined reference to `ME0GeometryBuilderFromCondDB::ME0GeometryBuilderFromCondDB()'
/cvmfs/cms-ib.cern.ch/nweek-02613/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: ME0GeometryESModule.cc:(.text+0x605): undefined reference to `ME0GeometryBuilderFromCondDB::build(RecoIdealGeometry const&)'
/cvmfs/cms-ib.cern.ch/nweek-02613/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: ME0GeometryESModule.cc:(.text+0x610): undefined reference to `ME0GeometryBuilderFromCondDB::~ME0GeometryBuilderFromCondDB()'
/cvmfs/cms-ib.cern.ch/nweek-02613/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: tmp/slc7_amd64_gcc820/src/Geometry/GEMGeometryBuilder/plugins/ME0GeometryESModule/ME0GeometryESModule.cc.o: in function `ME0GeometryESModule::produce(MuonGeometryRecord const&) [clone .cold.341]':
ME0GeometryESModule.cc:(.text.unlikely+0x31c): undefined reference to `ME0GeometryBuilderFromDDD::~ME0GeometryBuilderFromDDD()'
/cvmfs/cms-ib.cern.ch/nweek-02613/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: ME0GeometryESModule.cc:(.text.unlikely+0x368): undefined reference to `ME0GeometryBuilderFromCondDB::~ME0GeometryBuilderFromCondDB()'
/cvmfs/cms-ib.cern.ch/nweek-02613/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: tmp/slc7_amd64_gcc820/src/Geometry/GEMGeometryBuilder/plugins/ME0GeometryESModule/ME0GeometryESModule.cc.o: in function `edm::eventsetup::ProxyArgumentFactoryTemplate<edm::eventsetup::CallbackProxy<edm::eventsetup::Callback<ME0GeometryESModule, std::unique_ptr<ME0Geometry, std::default_delete<ME0Geometry> >, MuonGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<MuonGeometryRecord> >, MuonGeometryRecord, std::unique_ptr<ME0Geometry, std::default_delete<ME0Geometry> > >, edm::eventsetup::Callback<ME0GeometryESModule, std::unique_ptr<ME0Geometry, std::default_delete<ME0Geometry> >, MuonGeometryRecord, edm::eventsetup::CallbackSimpleDecorator<MuonGeometryRecord> > >::makeKey(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
ME0GeometryESModule.cc:(.text._ZNK3edm10eventsetup28ProxyArgumentFactoryTemplateINS0_13CallbackProxyINS0_8CallbackI19ME0GeometryESModuleSt10unique_ptrI11ME0GeometrySt14default_deleteIS6_EE18MuonGeometryRecordNS0_23CallbackSimpleDecoratorISA_EEEESA_S9_EESD_E7makeKeyERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZNK3edm10eventsetup28ProxyArgumentFactoryTemplateINS0_13CallbackProxyINS0_8CallbackI19ME0GeometryESModuleSt10unique_ptrI11ME0GeometrySt14default_deleteIS6_EE18MuonGeometryRecordNS0_23CallbackSimpleDecoratorISA_EEEESA_S9_EESD_E7makeKeyERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0xd): undefined reference to `char const* edm::typelookup::className<ME0Geometry>()'
/cvmfs/cms-ib.cern.ch/nweek-02613/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: ME0GeometryESModule.cc:(.text._ZNK3edm10eventsetup28ProxyArgumentFactoryTemplateINS0_13CallbackProxyINS0_8CallbackI19ME0GeometryESModuleSt10unique_ptrI11ME0GeometrySt14default_deleteIS6_EE18MuonGeometryRecordNS0_23CallbackSimpleDecoratorISA_EEEESA_S9_EESD_E7makeKeyERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZNK3edm10eventsetup28ProxyArgumentFactoryTemplateINS0_13CallbackProxyINS0_8CallbackI19ME0GeometryESModuleSt10unique_ptrI11ME0GeometrySt14default_deleteIS6_EE18MuonGeometryRecordNS0_23CallbackSimpleDecoratorISA_EEEESA_S9_EESD_E7makeKeyERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x16): undefined reference to `std::type_info const& edm::typelookup::classTypeInfo<ME0Geometry>()'
collect2: error: ld returned 1 exit status
gmake: *** [config/SCRAM/GMake/Makefile.rules:1742: tmp/slc7_amd64_gcc820/src/Geometry/GEMGeometryBuilder/plugins/ME0GeometryESModule/libME0GeometryESModule.so] Error 1
gmake: *** Waiting for unfinished jobs....
Copying tmp/slc7_amd64_gcc820/src/Geometry/HcalCommonData/src/GeometryHcalCommonData/libGeometryHcalCommonData.so to productstore area:
Copying tmp/slc7_amd64_gcc820/src/Geometry/HGCalCommonData/src/GeometryHGCalCommonData/libGeometryHGCalCommonData.so to productstore area:
Leaving library rule at src/DetectorDescription/DDCMS/plugins
Leaving library rule at src/DetectorDescription/DDCMS/plugins
Leaving library rule at Geometry/HGCalCommonData
Leaving library rule at Geometry/HcalCommonData
Leaving library rule at src/DetectorDescription/DDCMS/plugins
--- Registered EDM Plugin: RPCGeometryValidationPlugins
--- Registered EDM Plugin: DTGeometryValidationPlugins
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2

Reply:
this is because you do not link it against ```Geometry/GEMGeometryBuilder```

Reply:
@ianna : done